app/controllers/foods_controller.rb

```diff
@@ -0,0 +1,72 @@
+class FoodsController < ApplicationController
+  before_action :authenticate_user!
+
```

Comment:
The delete function is broken because it doesn't know what to delete. Add the line below to help Rails know what food item to delete.

`before_action :set_food, only: :destroy`

It helps to find the food by id and then proceeds with the code in the delete method.

Reply:
Hello @brytebee,
Correction taken and implemented.
Thank you